apis/apps/v1beta4/emqxbroker_webhook.go

```diff
@@ -233,3 +240,19 @@ func validateVolumeClaimTemplates(new, old Emqx) error {
 	}
 	return nil
 }
+
+func validateEmqxConfig(new, old Emqx) error {
+	oldEmqxConfig := old.GetSpec().GetTemplate().Spec.EmqxContainer.EmqxConfig
+	newEmqxConfig := new.GetSpec().GetTemplate().Spec.EmqxContainer.EmqxConfig
+	if oldEmqxConfig["name"] != newEmqxConfig["name"] {
+		return errors.New(`refuse to update the "name" field in ".spec.template.spec.emqxContainer.emqxConfig"`)
+	}
+	for k, oldValue := range oldEmqxConfig {
+		if strings.HasPrefix(k, "cluster") {
+			if newValue, ok := newEmqxConfig[k]; ok && newValue != oldValue {
+				return errors.New(`refuse to update the "^cluster.*" field in ".spec.template.spec.emqxContainer.emqxConfig"`)
```

Comment:
```suggestion
				return errors.New(`refuse to update the "^cluster.*$" field in ".spec.template.spec.emqxContainer.emqxConfig"`)
```

Reply:
> 

What is the significance of adding the $ sign, our string does not have an end constraint